framework/src/auxkernels/PenetrationAux.C

```diff
@@ -40,6 +40,9 @@ PenetrationAux::validParams()
   params.addParam<std::string>("normal_smoothing_method",
                                "Method to use to smooth normals (edge_based|nodal_normal_based)");
   params.addParam<MooseEnum>("order", orders, "The finite element order");
+  params.addCoupledVar("secondary_gap_offset", "offset to the gap distance from secondary side");
+  params.addCoupledVar("mapped_primary_gap_offset",
```

Comment:
I think I originally suggested the idea of including 'mapped' at the beginning of this parameter name. As I look at the usage of this, though, I'm not sure that it adds that much, and I think we should consider changing it to just `primary_gap_offset`.  There is no non-mapped version that we need to distinguish it from, right?

If you agree with changing the name, go ahead and do it.

Reply:
No. we will have a `primary_gap_offset` as an auxiliary variable to save the gap_offset from the primary side. We then map this variable to the secondary side. I think whenever we have a `mapped_primary_gap_offset`, we need to have a `primary_gap_offset` to map from.

Reply:
OK, I'll go for that. You really could name the gap offset AuxVariable that is defined on the primary surface something else, but I can go either way on this. I just thought it would simplify the input a bit.